app/graphql/queries/user.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Queries
+  class User < Queries::BaseQuery
+    description "get user by argument"
+    type Types::Models::UserType, null: false
+    argument :id, Integer, required: false
+    argument :login, String, required: false
+    argument :email, String, required: false
+    argument :name, String, required: false
+
+    def resolve(args)
+      return {} unless args
+
+      if args.key?(:id)
+        ::User.find(args[:id])
+      elsif args.key?(:login)
+        ::User.find_by(login: args[:login])
+      elsif args.key?(:email)
+        ::User.find_by(email: args[:email].sub(/ <.*>$/, ""))
```

Comment:
I'm too dense to understand this. Is this how users who are not in admin mode are prevented from finding a user by email?